.azurepipelines/templates/CargoToolRepublish.yml

```diff
@@ -0,0 +1,194 @@
+## @file
+# Azure Pipeline to build and publish cargo tools in a format supported by edk2-pytool-extension's external dependency.
+#
+# This template will clone the repository specified in the `repo_url` parameter at the tag specified in the `tag`
+# parameter and then build the tool for X64 and ARM64 for Windows and/or linux. These binaries will be published as a
+# nuget package in an internal format supported by edk2-pytool-extensions.
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+parameters:
+  - name: tag
+    displayName: 'Cargo Make Version Tag'
+    type: string
+  - name: repo_url
+    displayName: 'Cargo Tool Repository URL'
+  - name: tool_name
+    displayName: 'Tool name (and nuget config file name)'
+    type: string
+  - name: extra_build_args
+    displayName: 'Extra build arguments'
+    type: string
+    default: ''
+  - name: upload_linux
+    displayName: 'Upload Linux Binaries (X64 and ARM64)'
+    type: boolean
+    default: true
+  - name: upload_windows
+    displayName: 'Upload Windows Binaries (X64 and ARM64)'
+    type: boolean
+    default: true
+  - name: python_version
+    displayName: 'Python Version'
+    type: string
+    default: '3.11'
+
+variables:
+  - name: windows_image
+    value: 'windows-latest'
+  - name: linux_image
+    value: 'ubuntu-22.04'
+
+stages:
+- stage: build_binaries
+  displayName: 'Build ${{ parameters.tool_name }} Artifact(s)'
+
+  jobs:
+    - job: build_linux
+      displayName: Build Linux Binaries
```

Comment:
It looks like these Linux & Windows build sections might be a little cleaner with a matrix from aarch64 and x86_64. But there's not much as-is either.

Reply:
Things are different enough between the two pipelines that I decided to have two separate jobs rather than a pipeline. 

There differences are mainly:

1. different architectures downloaded per OS
2. Linux uses `cross` to compile to reduce complexity of cross compilation on linux
3. linux makes the binaries execute

I can look into making it a matrix if you want, but it may not be worth the time it will take to make it work

Reply:
I think it's okay as-is. It's not something I'd expect to benefit tremendously from the matrix anyway in terms of future extension.